src/tbp/monty/frameworks/models/object_model.py

```diff
@@ -794,20 +801,31 @@ def _get_new_voxel_features(
                 avg_feat = get_most_common_bool(feats)
                 # NOTE: object_id may need its own most common function until
                 # IDs actually represent similarities
+            elif feature == "object_id":
+                avg_feat = get_most_common_value(feats)
             else:
                 avg_feat = np.mean(feats, axis=0)
             # Only take average if there was a feature stored here before.
             # since self._observation_count already includes the new obs
             # this needs to be > the number of new feature obs in the voxel.
-            if self._observation_count[voxel[0], voxel[1], voxel[2], 0] > len(feats):
+            num_obs_in_voxel = self._observation_count[voxel[0], voxel[1], voxel[2], 0]
+            num_new_obs = len(feats)
+            if num_obs_in_voxel > num_new_obs:
```

Comment:
I like that the two parts were separated out into named variables, as it makes this easier to understand.